src/python/pants/backend/native/subsystems/xcode_cli_tools.py

```diff
@@ -133,6 +134,12 @@ def include_dirs(self):
       if is_readable_dir(secure_inc_dir):
         all_inc_dirs.append(secure_inc_dir)
 
+      # TODO: figure out why we need to specify this directory separately.
+      if include_cpp_inc:
+        cpp_inc_dir = os.path.join(d, 'c++/v1')
```

Comment:
'c++/v1' seems like dark magic. Does it rely on xcode version x.y.z? Is it always available? What the heck does it signify, something like c++11 - is this a c++ standards committee moniker? ... If there is a concise pointer to doc that defines this dir, that would be ideal.

Reply:
I was actually able to remove this as a result of the other changes and testing added in this PR. I additionally left a pointer to #6143, which would cover potentially removing the rest of the sprawling attempt at discovering hardcoded paths.